### DISCUSSION TITLE
cURL SSL certificate problem

### DISCUSSION THREAD

**Discussion by DavidKrenekCZ** [statamic/cms#4677] [Q&A] [answered]
Hi guys, on site with Statamic v2 we have a problem with generating images, especially with cURL SSL certificate. Images generating stopped working because of expiration Let's Encrypt's root certificate. There are more details in the attached picture. Our hosting provider recommends pasting somewhere code bellow, but I don't know where. I've tested root `index.php`, but it doesn't work. Could somebody help me, please? Or is there some other way to fix that issue?

```
curl_setopt( $curl, CURLOPT_CAINFO, '/global/applications/tools/cacert.pem');
curl_setopt( $curl, CURLOPT_CAPATH, '/global/applications/tools/');
curl_setopt( $curl, CURLOPT_PORT , 443);
```
<img width="1145" alt="Snímek obrazovky 2021-11-08 v 10 14 11" src="https://user-images.githubusercontent.com/22493634/140715517-0111fde7-77db-4296-ab48-6fe47c2aba4f.png">

![cURL](https://user-images.githubusercontent.com/22493634/140715622-9053ac41-4448-43e5-afc1-69247c1e11df.png)

**Comment by jasonvarga** [comment]
I've never heard of this error on a production server. Locally, yes, where you might want to use ssl on your development sites.
If there's an SSL issue on production I would think it's a server configuration issue. I don't think you should need to add PHP code.

**Comment by DavidKrenekCZ** [comment]
Thank you Jason. I think so, but hosting provider advises me to add this. In other way, don't you know if is there a possibility to temporary turn off SSL in this case and made connection only by http?

**Comment by jasonvarga** [comment, marked as answer]
Actually looking at the stack trace, your glide tag is using the "external url" method of generating an image.

Are you actually trying to generate an image for an external url?

If it's one of your own assets, try to pass the appropriate url to it. Then it won't even try to make a curl request.
What's the template containing the glide tag?

**Comment by DavidKrenekCZ** [comment]
You was right. Although it was local asset, it was loaded by absolute url. I changed it to relative and that fixes the problem - cURL is no more needed. Thank you very much Jason!

<img width="764" alt="Snímek obrazovky 2021-11-12 v 13 17 28" src="https://user-images.githubusercontent.com/22493634/141466975-9bbc9e98-ccc9-4d0b-a7ea-710023db6825.png">
